### PR TITLE
Update Rust demo and dependencies

### PR DESCRIPTION
**Pull request by zulamdat**
Asana: https://app.asana.com/0/1206389362636911/1207189648450385/f

Changes:
* Update Rust Demo to latest query improvement
* Update test data set

How to test:
Note: test this PR in your local until further notice
1. Go to rust directory
2. build with `cargo build -r`
3. add `export LD_LIBRARY_PATH=./lib`
4. Run the demo with `./target/release/zkpass-demo r0 test/data/dewi-profile.json test/data/bca-insurance-dewi-dvr.json`
5. Expected result
```bash
...
#### verification completed []
json-result={
  "name": "Dewi",
  "result": true
}
>> output list:
key=name, value=Str("Dewi")
key=result, value=Bool(true)
<< end of list
the query result is true
```

### CODE REVIEW

File: rust/zkpass-demo/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "zkpass-demo"
-version = "0.2.0-beta.2"
+version = "0.3.0-rc.1"
```
Comment (LawrencePatrickSianto) [inline rust/zkpass-demo/Cargo.toml:3]: memang `0.3.0-rc.1` kah? nda ikut ke `1.0.0-alpha.1`?

**Comment by LawrencePatrickSianto** [comment]
apa perlu update rust doc juga di PR ini?

File: rust/zkpass-demo/Cargo.toml (hunk 1 above)
Comment (zulamdat) [reply rust/zkpass-demo/Cargo.toml:3]: boleh gw update ya

**Comment by zulamdat** [comment]
> apa perlu update rust doc juga di PR ini?

kayanya dibikin terpisah aja

File: rust/zkpass-demo/Cargo.toml (hunk 1 above)
Comment (zulamdat) [reply rust/zkpass-demo/Cargo.toml]: done update, jadinya gw update ke v1.0.0 sekalian ya

File: rust/README.md [hunk 2]
```diff
@@ -2,38 +2,50 @@
...
 </pre>
 
+## Run the demo
 
+Follow these steps to test demo:
 
+1. In rust directory
+1. Build zkPass with `cargo build --release`
+1. Add environment with command `export LD_LIBRARY_PATH=./lib`
+1. Run the demo with `./target/release/zkpass-demo r0 test/data/dewi-profile-wrong.json test/data/bca-insurance-dewi-dvr.json`
```
Comment (LawrencePatrickSianto) [inline rust/README.md:49]: sebaiknya urut angka 1,2,3,4

Comment (zulamdat) [reply rust/README.md]: otomatis dari formatter law, nanti di previewnya tetep urut
image
tapi tetep dah gw update yaa

Comment (LawrencePatrickSianto) [reply rust/README.md]: i see, okay

**Review by LawrencePatrickSianto — APPROVED** [review]

File: rust/zkpass-client/src/zkpass_client.rs [hunk 3]
```diff
@@ -181,24 +185,20 @@ impl ZkPassProofGenerator for ZkPassClient {
...
-            Ok(response_body)
+            let proof = response_body["proof"].as_str().unwrap().to_string();
+            Ok(proof)
         } else {
-            Err(
-                ZkPassError::CustomError(
-                    format!("Request failed with status code: {}", response.status())
-                )
-            )
+            Err(ZkPassError::CustomError(format!("Request failed: {}", response_body.to_string())))
```
Comment (WilliamhGDP) [inline rust/zkpass-client/src/zkpass_client.rs:201]: line 188 - 201 jgn lupa di keepup ama zkpass_client di zkpass repo, untuk mencegah 408 response
```

        let is_response_success = response.status().is_success();
        let response_body = match response.text().await {
            Ok(body) => body,
            Err(e) => {
                return Err(ZkPassError::CustomError(e.to_string()));
            }
        };

        if is_response_success {
            let response_body: Value = serde_json::from_str(&response_body).unwrap();
            // extract the proof under the root "proof" element
            let proof = response_body["proof"].as_str().unwrap().to_string();
            Ok(proof)
        } else {
            Err(ZkPassError::CustomError(format!("Request failed: {}", response_body.to_string())))
        }
```

File: rust/zkpass-demo/src/proof_verifier.rs [hunk 4]
```diff
@@ -85,22 +88,20 @@ impl ProofVerifier {
     //  Simulating the Proof Verifier's get_dvr_token REST API
     //
     pub fn get_dvr_token(&self, zkvm: &str, dvr_file: &str) -> String {
-        const VERIFIER_PRIVKEY: &str = r"-----BEGIN PRIVATE KEY-----
+        const VERIFIER_PRIVKEY: &str =
```
Comment (WilliamhGDP) [inline rust/zkpass-demo/src/proof_verifier.rs:91]: klo dari development zkpass repo,
`ISSUER_PRIVKEY` dan `VERIFIER_PRIVKEY` udah dipisah ke 1 file (sample.keys.rs)

File: rust/zkpass-client/src/zkpass_client.rs (hunk 3 above)
Comment (WilliamhGDP) [reply rust/zkpass-client/src/zkpass_client.rs:201]: Nanti keepup ama ini ya:
1. https://github.com/GDP-ADMIN/zkPass/pull/276
2. https://github.com/GDP-ADMIN/zkPass/pull/266

Comment (zulamdat) [reply rust/zkpass-client/src/zkpass_client.rs:207]: Updated ya https://github.com/gl-zkPass/zkpass-sdk/pull/63/commits/454bc018afb3f3f6c072e678222483154db1d312

File: rust/zkpass-demo/src/proof_verifier.rs (hunk 4 above)
Comment (zulamdat) [reply rust/zkpass-demo/src/proof_verifier.rs]: updated ya, jadinya gw taro di constant aja, please check https://github.com/gl-zkPass/zkpass-sdk/pull/63/commits/31d6516db1a785cab7a48083c43c3e4c83bccabf

**Review by WilliamhGDP — APPROVED** [review]

**Review by pp-pal — APPROVED** [review]

**Pull request merged**